test/decl/enum/enumtest.swift

```diff
@@ -636,3 +636,9 @@ if case nil = foo1 {} // Okay
 if case .none? = foo1 {} // Okay
 if case nil = foo2 {} // Okay
 if case .none?? = foo2 {} // Okay
+
+enum EnumCaseWithGenericDeclaration {
+  case foo<T>(T) // expected-error {{generic signature cannot be declared in enum 'case'. did you mean to attach it to enum declaration?}}
```

Comment:
Can we offer the second clause here as a fixit that reparents the generic parameters into the parent enum declaration?

Reply:
We probably can, I just have no idea how to implement this change however - will take a better look to see if I can come up with something.

Reply:
Done.